packages/vsce/src/SshConfigUtils.ts

```diff
@@ -23,27 +32,105 @@ export class SshConfigUtils {
+        const zoweExplorerApi = ZoweVsCodeExtension.getZoweExplorerApi();
+        const profCache = zoweExplorerApi.getExplorerExtenderApi().getProfilesCache();
+
         if (profileName != null) {
             return profCache.getLoadedProfConfig(profileName, "ssh");
         }
-        const sshProfiles = (await profCache.fetchAllProfilesByType("ssh")).filter(
-            ({ name, profile }) => name != null && profile?.host,
+
+        const sshProfiles = (await profCache.fetchAllProfilesByType("ssh"))
+            .filter(({ name, profile }) => name != null && profile?.host);
+
+        // get configs via migrateSshConfig() and remove options that have host name that already exists as an ssh profile on the config file
+        const migratedConfigs = (await SshConfigUtils.migrateSshConfig()).filter(migratedConfig =>
+            !sshProfiles.some(sshProfile => sshProfile.profile?.host === migratedConfig.hostname)
         );
+
         const qpItems: vscode.QuickPickItem[] = [
-            ...sshProfiles.map((prof) => ({ label: prof.name!, description: prof.profile?.host })),
             { label: "$(plus) Add New SSH Host..." },
+            ...sshProfiles.map((prof) => ({ label: prof.name!, description: prof.profile?.host })),
+            { label: "Merge From SSH Config", kind: vscode.QuickPickItemKind.Separator },
+            ...migratedConfigs.map((config) => ({ label: config.name!, description: config.hostname }))
         ];
+
         const result = await Gui.showQuickPick(qpItems, { title: "Choose an SSH host" });
-        if (result === qpItems[qpItems.length - 1]) {
-            if (await SshConfigUtils.createTeamConfig()) {
-                return profCache.getDefaultProfile("ssh");
+
+        let selectedProfile: sshConfigExt | undefined;
+
+        if (result === qpItems[0] || migratedConfigs.some((config) => result?.label === config.name!)) {
+            if(result === qpItems[0]){
+                // if (await SshConfigUtils.createTeamConfig()) {
+                //     return profCache.getDefaultProfile("ssh");
+                // }
+                selectedProfile = await SshConfigUtils.createNewConfig();
+                await profCache.refresh();
+            }
+            else if(migratedConfigs.some((config) => result?.label === config.name!)){
+                selectedProfile = migratedConfigs.find((config) => result!.label === config.name!);
             }
-        } else if (result != null) {
+            if (selectedProfile) {
+                selectedProfile = await SshConfigUtils.getNewProfileName(selectedProfile);
+                selectedProfile = await SshConfigUtils.promptForAuth(selectedProfile);
+            }
+        }
+        else if (result != null) {
             return sshProfiles.find((prof) => prof.name === result.label);
         }
     }
 
+    private static async migrateSshConfig(): Promise<sshConfigExt[]> {
```

Comment:
Should we move this method to the SDK package and make it a public method there so that it can be invoked by extenders outside of the VSCE?